src/loaddb/load_object.c

```diff
@@ -486,11 +486,11 @@ text_print (TEXT_OUTPUT * tout, const char *buf, int buflen, char const *fmt, ..
       va_end (ap);
     }
 
-  if (nbytes > 0 && buf != NULL)
+  if (nbytes > 0)
     {
       if (nbytes < size)
 	{			/* OK */
-	  if (buflen > 0)
+	  if (buflen > 0 && buf != NULL)
 	    {			/* unformatted print */
 	      memcpy (tout->ptr, buf, buflen);
```

Comment:
It seems to change the existing semantic. The changed code skips `memcpy()` if buf  is NULL, but it just assumed `buf` is not NULL if `buflen > 0` in the previous code. 
```suggestion
	  if (buflen > 0)
	  {
	     assert (buf != NULL)
```